src/AutoRest.CSharp.V3/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -80,10 +87,15 @@ private static bool CanBeInitializedInline(Parameter parameter)
                 return true;
             }
 
+            if (TypeFactory.IsStruct(parameter.Type) && parameter.DefaultValue.Value.Value != null)
+            {
+                return false;
+            }
+
             return parameter.Type.IsValueType || parameter.DefaultValue.Value.Value == null;
         }
 
-        private static bool CanWriteNullCheck(Parameter parameter) => parameter.ValidateNotNull && (parameter.Type.IsNullable || !parameter.Type.IsValueType);
+        private static bool CanWriteNullCheck(Parameter parameter) => parameter.ValidateNotNull && (parameter.Type.IsNullable || !parameter.Type.IsValueType) && parameter.DefaultValue == null;
```

Comment:
You might want to remove a similar condition from `HasNullCheck` so they are not duplicated.

Reply:
Conditions are not duplicated `parameter.DefaultValue != null` is needed in order to call this `!TypeFactory.CanBeInitializedInline(parameter.Type, parameter.DefaultValue)` in `HasNullCheck`.